Bug: test_dequeue_ecn fails on Broadcom 7050CX3: last-packet ECN assertion is platform-dependent

### Is it platform specific

broadcom

### Importance or Severity

Medium

### Description of the bug

### Summary

`tests/snappi_tests/ecn/test_dequeue_ecn_with_snappi.py::test_dequeue_ecn` fails on Broadcom 7050CX3 SKUs

```python
pytest_assert(not is_ecn_marked(ip_pkts[-1]),
              "The last packet should not be marked")
```

The test passes on Mellanox SKUs and fails on Broadcom 7050CX3 not because of any DUT/WRED defect, but because the assertion is implicitly tied to per-SKU buffer geometry.



### Steps to Reproduce

### Reproduction

- Testbed: any Broadcom 7050CX3-based RDMA testbed
- Command: `pytest tests/snappi_tests/ecn/test_dequeue_ecn_with_snappi.py::test_dequeue_ecn`
- Test parameters (current defaults): `kmin=50000`, `kmax=51000`, `pmax=100`, `data_flow_pkt_size=1024`, `data_flow_pkt_count=101`



### Actual Behavior and Expected Behavior

### Observed failure

```
AssertionError: The last packet should not be marked
```

<img width="1315" height="822" alt="Image" src="https://github.com/user-attachments/assets/aec6427d-243a-4e55-a0f4-ab5e3bae0ad8" />

### Root cause

There are two problems:

1. WRED threshold quantization. On Tomahawk(/Trident) the WRED min/max are programmed in bytes but quantized internally to MMU cells / threshold units (256-byte cells on TH/TH2/TH3, plus an additional power-of-2/index quantization). 50000 and 51000 round to the same internal threshold bucket, so you effectively get kmin == kmax. There's no marking-probability ramp — once depth ≥ that value, marking probability is 100%, and below it, 0%. There is no "tail" where pmax linearly falls.

2. MMU/buffer accounting unit and dequeue rate. Tomahawk accounts queue occupancy at the egress queue in MMU cells; with shared-pool dynamics + PFC release behavior, the queue stays at or above the (single) threshold bucket for essentially the whole 101-packet dequeue window. So every packet — including ip_pkts[-1] — sees depth ≥ kmax and is CE-marked. → assertion not is_ecn_marked(ip_pkts[-1]) fails.

You can see this directly in the failure log: original Broadcom WRED profile had green_min=1,048,576 / green_max=2,097,152 (1–2 MB), confirming the platform expects threshold values orders of magnitude larger and well-separated.

### Affected files
- test_dequeue_ecn_with_snappi.py

### Affected platforms
- Broadcom 7050CX3 (confirmed)

### Related PR



### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/snappi_tests/ecn/test_dequeue_ecn_with_snappi.py
+++ b/tests/snappi_tests/ecn/test_dequeue_ecn_with_snappi.py
@@ -26,7 +26,7 @@ pytestmark = [pytest.mark.topology('multidut-tgen', 'tgen')]
 # sent to see the marked packets at the end of the flow.
 ECN_PARAMS_BY_ASIC = {
     'default':  {'ecn_params': {'kmin': 50000, 'kmax': 51000, 'pmax': 100}, 'pkt_count': 101},
-    'broadcom': {'ecn_params': {'kmin': 40000, 'kmax': 51000, 'pmax': 100}, 'pkt_count': 301},
+    'broadcom': {'ecn_params': {'kmin': 40000, 'kmax': 168000, 'pmax': 100}, 'pkt_count': 301},
 }
 
 

